framework/src/main/java/org/checkerframework/framework/util/defaults/QualifierDefaults.java

```diff
@@ -919,36 +914,24 @@ protected class DefaultApplierElement {
         /** The default element applier implementation. */
         protected final DefaultApplierElementImpl impl;
 
-        /**
-         * Local type variables are defaulted to top when flow is turned on We only want to default
-         * the top level type variable (and not type variables that are nested in its bounds). E.g.,
-         * {@code <T extends List<E>, E extends Object> void method() { T t; } }.
-         *
-         * <p>We would like t to have its primary annotation defaulted but NOT the E inside its
-         * upper bound. we use referential equality with the top level type var to determine which
-         * ones are definite type uses, i.e. uses which can be defaulted
-         */
-        private final @Nullable AnnotatedTypeVariable defaultableTypeVar;
-
         /**
          * Create an instance.
          *
          * @param atypeFactory the type factory
          * @param scope the scope for the defaults
          * @param type the type to default
-         * @param applyToTypeVar whether to apply defaults to type variable uses
          */
         public DefaultApplierElement(
-                AnnotatedTypeFactory atypeFactory,
-                Element scope,
-                AnnotatedTypeMirror type,
-                boolean applyToTypeVar) {
+                AnnotatedTypeFactory atypeFactory, Element scope, AnnotatedTypeMirror type) {
             this.atypeFactory = atypeFactory;
             this.qualHierarchy = atypeFactory.getQualifierHierarchy();
             this.scope = scope;
             this.type = type;
+            this.shouldDefaultTypeVarLocals =
```

Comment:
In the previous version, I set this at the outer level, which stored an incorrect value.
The value of `GATF#getShouldDefaultTypeVarLocals()` changes when determining the LHS type of a local variable.
This change made the `all-systems` test pass again. The `RegexUtil` suppression is still necessary.
If no other tests fail, that seems acceptable.

Reply:
`./gradlew checkResourceLeak` still gives errors like this:

````
dataflow/src/main/java/org/checkerframework/dataflow/analysis/ForwardAnalysisImpl.java:535: error: [builder:required.method.not.known] The checker cannot determine the must call methods of newThenStore or any of its aliases, so it could not determine if they were called. Typically, this error indicates that you need to write an @MustCall annotation (often on an unconstrained generic type).
                    S newThenStore = mergeStores(s, thenStore, shouldWiden);
                      ^
  The type of object is: S.
  Reason for going out of scope: regular method exit
````

I tried adding `@MustCall` to the S type parameter bound, but get the same failures.

@Ao-senXiong Can you see whether your previous test cases now pass and then see what the problem with the Resource Checker is?